web/src/app/pages/create-survey/task-details/task-input/task-input.component.ts

```diff
@@ -0,0 +1,354 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter,
+  OnChanges,
+  SimpleChanges,
+  OnDestroy,
+  ViewChildren,
+  QueryList,
+  HostListener,
+  ElementRef,
+  ViewChild,
+  ChangeDetectorRef,
+} from '@angular/core';
+import {
+  FormGroup,
+  FormBuilder,
+  Validators,
+  FormControl,
+  AbstractControl,
+  ValidationErrors,
+} from '@angular/forms';
+import { DialogService } from 'app/services/dialog/dialog.service';
+import { TaskType } from 'app/models/task/task.model';
+import { Option } from 'app/models/task/option.model';
+import { List } from 'immutable';
+import {
+  MultipleChoice,
+  Cardinality,
+} from 'app/models/task/multiple-choice.model';
+import { CdkDragDrop } from '@angular/cdk/drag-drop';
+import { JobService } from 'app/services/job/job.service';
+import { firstValueFrom, Subscription } from 'rxjs';
+import { TaskGroup } from '../task-details.component';
+
+
+export interface TaskTypeSelectOption {
+  icon: string;
+  label: string;
+  type: TaskType;
+  cardinality?: Cardinality;
+}
+
+@Component({
+  selector: 'task-input',
+  templateUrl: './task-input.component.html',
+  styleUrls: ['./task-input.component.scss'],
+})
+export class TaskInputComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() label?: string;
+  @Input() required?: boolean;
+  @Input() taskType?: TaskType;
+  @Input() multipleChoice?: MultipleChoice;
+  @Input() cardinality?: Cardinality;
+  @Input() taskCount?: Number;
+  @Output() update = new EventEmitter();
+  @Output() delete = new EventEmitter();
+  taskOptions: MultipleChoice | undefined;
+  selectTaskOptions: TaskTypeSelectOption[];
+  @Input() taskIndex?: number;
+  // TODO: This name TaskGroup is ambigous. We need to resolve this in
```

Comment:
Please add a tracking issue clarifying this and reference as `TODO(#nnn):..`. 

Nit: s/ambigous/ambiguous/

Reply:
Done!